### PR TITLE
Upgrade Jackson version to fix high vuln

### PR DESCRIPTION
**Pull request by paulmr**
## What does this change?

This upgrades the Jackson version due to a high priority security vulnerability.

## How to test

As this is a library, it needs a client application to test. I tried it with [pubflow](https://github.com/guardian/pubflow/tree/pmr/firehose-client-preview-release) on `CODE` and it seems to work fine (i.e. it boots and displays updates to content).

## How can we measure success?

This should be unnoticable.

## Have we considered potential risks?

Should be OK as it will involve releasing a new version.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -38,14 +38,14 @@ resolvers ++= Resolver.sonatypeOssRepos("releases")
 libraryDependencies ++= Seq(
   "com.gu" %% "content-api-models-scala" % "25.0.0",
   "com.gu" %% "thrift-serializer" % "5.0.5",
-  "software.amazon.kinesis" % "amazon-kinesis-client" % "2.5.6",
```
Comment (paulmr) [inline build.sbt:41]: upgrading this to the latest version just for the sake of it really.

**Review by emdash-ie — APPROVED** [review]
Testing by deploying to pubflow-CODE sounds sufficiently thorough, so I’m happy to approve. I’m not 100% certain pubflow will exercise every path that might break, but that’s because I’m unfamiliar with pubflow rather than because I have a specific case in mind.

It doesn’t like there are any existing tests for this library, unless I misunderstand? That’s a shame: we should consider adding some in a subsequent PR.

**Comment by paulmr** [comment]
> It doesn’t like there are any existing tests for this library, unless I misunderstand? That’s a shame: we should consider adding some in a subsequent PR.

thanks emily. Yes, you are right, tests would be great. I wonder what's the best way to achieve that, given that it's sole purpose is to talk to the kinesis stream - perhaps we could mock the stream somehow?

**Comment by emdash-ie** [comment]
> > It doesn’t like there are any existing tests for this library, unless I misunderstand? That’s a shame: we should consider adding some in a subsequent PR.
> 
> thanks emily. Yes, you are right, tests would be great. I wonder what's the best way to achieve that, given that it's sole purpose is to talk to the kinesis stream - perhaps we could mock the stream somehow?

I hadn’t really thought about what form the tests could/would need to take. Perhaps something a little like [the ophan e2e tests](https://github.com/guardian/ophan/blob/main/e2e-test/readme.md)? It would be nice to be able to make assertions like “given this content on the kinesis stream, an example ContentApiFirehoseConsumer would receive these calls”.

**Pull request merged**